pylint/checkers/base/basic_checker.py

```diff
@@ -367,10 +367,11 @@ def _check_using_constant_test(
                 # Just forcing the generator to infer all elements.
                 # astroid.exceptions.InferenceError are false positives
                 # see https://github.com/pylint-dev/pylint/pull/8185
-                if isinstance(inferred, nodes.FunctionDef):
-                    call_inferred = list(inferred.infer_call_result(node))
-                elif isinstance(inferred, nodes.Lambda):
-                    call_inferred = list(inferred.infer_call_result(node))
+                match inferred:
+                    case nodes.FunctionDef():
+                        call_inferred = list(inferred.infer_call_result(node))
+                    case nodes.Lambda():
+                        call_inferred = list(inferred.infer_call_result(node))
```

Comment:
```suggestion
                    case nodes.FunctionDef() | nodes.Lambda():
                        call_inferred = list(inferred.infer_call_result(node))
```

Reply:
Good catch, though the match statement isn't really necessary here at all then. Replaced it with
```py
if isinstance(inferred, (nodes.FunctionDef, nodes.Lambda)):
    ...
```